updatecli/updatecli.d/klipper-helm-and-controller.yaml

```diff
@@ -0,0 +1,114 @@
+---
+name: "Bump Klipper Helm and Helm Controller version"
+scms:
+  k3s:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ .github.username }}"
+      token: "{{ requiredEnv .github.token }}"
+      owner: "{{ .k3s.org }}"
+      repository: "{{ .k3s.repo }}"
+      branch: "{{ .k3s.branch }}"
+      commitmessage:
+        title: "Bump Klipper Helm and Helm Controller version"
+  helm-controller:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ .github.username }}"
+      token: "{{ requiredEnv .github.token }}"
+      owner: "{{ .helm_controller.org }}"
+      repository: "{{ .helm_controller.repo }}"
+      branch: "{{ .helm_controller.branch }}"
+
+actions:
+  github:
+    title: "Bump Klipper Helm and Helm Controller version"
+    kind: "github/pullrequest"
+    scmid: "k3s"
+    spec:
+      automerge: false
+      mergemethod: "squash"
+      usetitleforautomerge: true
+      parent: false
+      labels: 
+        - "dependencies"
+
+sources:
+  helm-controller:
+    name: "Get Helm Controller latest release version"
+    kind: "githubrelease"
+    spec:
+      owner: "{{ .helm_controller.org }}"
+      repository: "{{ .helm_controller.repo }}"
+      branch: "{{ .helm_controller.branch }}"
+      token: "{{ requiredEnv .github.token }}"
+      versionfilter:
+        kind: "latest"
+  klipper-helm:
+    name: "Get Klipper Helm latest release version"
+    kind: "githubrelease"
+    spec:
+      owner: "{{ .klipper_helm.org }}"
+      repository: "{{ .klipper_helm.repo }}"
+      branch: "{{ .klipper_helm.branch }}"
+      token: "{{ requiredEnv .github.token }}"
+      versionfilter:
+        kind: "latest"
+  get-pwd:
+    name: "Run Updatecli execution directory"
+    kind: "shell"
+    disablesourceinput: true
+    spec:
+      command: 'pwd'
+      environments:
+        - name: PATH
+
+conditions:
+  klipper-helm:
+    name: "Check rancher/klipper-helm image version in DockerHub"
+    kind: "dockerimage"
+    sourceid: "klipper-helm"
+    spec:
+      image: "rancher/klipper-helm"
+  helm-controller:
+    name: "Check Helm Controller usage in go.mod"
+    kind: "file"
+    scmid: "k3s"
+    disablesourceinput: true
+    spec:
+      file: "go.mod"
+      matchpattern: 'github.com/k3s-io/helm-controller'
+  klipper-helm-in-helm-controller:
+    name: "Check rancher/klipper-helm image version in Helm Controller"
+    kind: "file"
+    scmid: "helm-controller"
+    disablesourceinput: true
+    spec:
+      file: "pkg/controllers/chart/chart.go"
+      matchpattern: 'rancher/klipper-helm:{{ source "klipper-helm" }}'
```

Comment:
Just to make sure I get it:
```
if 
    dockerhub has the latest klipper-helm version
&&
    the k3s go.mod includes k3s-io/helm-controller
&&
    the latest helm-controller has the latest klipper-helm version
then
    don't update
else
    run the 'run-go-mod-update' script passing in the latest helm-controller version
and
    update the k3s airgap image-list to have the latest klipper-helm version
```

Reply:
That's right! Had to do this way, in order to make sure that (1) `the latest helm-controller has the latest klipper-helm version`, so then we can run the `go mod` and `update the k3s airgap image-list to have the latest klipper-helm version` from (1).

Reply:
Shouldn't we check that the helm-controller version in the go.mod matches the latest version?

Reply:
You mean checking in here?

https://github.com/k3s-io/k3s/blob/d388b82d252d1b900527b1fd9a60706e43c77f26/go.mod#L109

It happens as a target condition:

https://github.com/k3s-io/k3s/blob/1d84e39f3eb04607408c15335da6e59a8d13fc3d/updatecli/updatecli.d/klipper-helm-and-controller.yaml#L101

Reply:
sorry, to be more clear:
I wonder if you meant to use the "disablesourceinput" parameter in this condition, if you left it out the condition would validate the version of the helm-controller in the go.mod like `github.com/k3s-io/helm-controller <input>`, currently it only checks the existence of "helm-controller" in the go.mod like `github.com/k3s-io/helm-controller`

I am just making sure this was a choice and not a typo.

Reply:
Ohh, got it. You mean in here - https://github.com/k3s-io/k3s/pull/7146/files#diff-b541f33a50cd734727c9018f7da3aaf13a3c2721a1033b1fe2ca622cc314428fR77-R84, right?

My reasoning was really to set `disablesourceinput: true` because if the version being used of `github.com/k3s-io/helm-controller` in `go.mod` is older than the current last version available, it will not fail the condition, instead it will go ahead and do the bump in the target step.

Apologies in advance if I misunderstood your question again.

Reply:
np, glad we are on the same page now, thanks!